infrastructure/api/handler/login_handelr.go

```diff
@@ -0,0 +1,43 @@
+package handler
+
+import (
+	"log"
+
+	"tat_gogogo/infrastructure/middleware"
+	"tat_gogogo/interface/controller"
+
+	"github.com/gin-gonic/gin"
+)
+
+/*
+LoginHandler is a function for gin to handle login api
+*/
+func LoginHandler(c *gin.Context) {
+	authMiddleware, err := middleware.NewAuthMiddleware()
+	if err != nil {
+		c.Status(500)
+		log.Fatal("JWT Error:" + err.Error())
```

Comment:
`log.Fatal` will crash your code. According to their [documentation](https://golang.org/pkg/log/#Fatal), it will `log.Print` then it will `os.Exit(1)`. You shouldn't crash your code in the case of a failed Middleware Authentication

Reply:
so in here I just use the regular print?

Reply:
Your common `log.Printf` will work, and whenever you **purposefully** want to break the execution of your code, use `panic` or `os.Exit(1)`

Reply:
I've fix it with regular print